### PR TITLE
[WIP] Trying to get integration testing to test against cockpit.dbus API

### PR DESCRIPTION
**Pull request by rarebreed**
This is currently not working, but I wanted to show what was being done.  I have [linked a screenshot](http://imgur.com/a/iRset) of what happens when the DBusSpecRunner.html page is run (from the left hand side bar, you should see a plugin named DBus Testing which will load this page).

As a side note about having a single entry point.  This doesn't seem possible.  The reason is that the cockpit rpm only installs the gzipped/minified version of the base1/cockpit.js file.  Doing 

```javascript
import cockpit from "../base1/cockpit";

// this only works because cockpit was loaded via the <script src="../base1/cockpit,js"> tag
const cockpit = require("cockpit");
```
Does not work.  This is why if you look at all the existing examples of plugins in cockpit, they do not (solely) import or require the cockpit.js file.  There is always a <script> tag that loads the cockpit file.  This is because the <script> tag can decompress the gzipped file.

This in turn makes a single entry point hard or impossible.  For example, I can not create dbus.test.js which does an import (without a <script> load) and expect webpack to find this dependency...since it doesn't exist.

### CODE REVIEW

**Comment by dperpeet** [comment]
> Does not work. This is why if you look at all the existing examples of plugins in cockpit, they do not (solely) import or require the cockpit.js file. There is always a <script> tag that loads the cockpit file. This is because the <script> tag can decompress the gzipped file.

To be clear, this works when connected to the cockpit server because cockpit-ws decompresses the file and serves it to the client.

**Comment by dperpeet** [comment]
In order to interface with DBUS via the Cockpit API, Cockpit needs to be running.

**Comment by kahowell** [comment]
Can't we just add jasmine via NPM instead of adding it to the repo?

**Comment by kahowell** [comment]
I did a bit of work in #1687 to integrate jasmine for both unit and functional tests. See the PR for details. I took some of the work from this PR and incorporated it there. Please take a look.

File: cockpit/manifest.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (kahowell) [inline cockpit/manifest.json]: I found a way to do this via override.json. See #1687 for details :smile:

File: cockpit/package.json [hunk 2]
```diff
@@ -10,20 +10,29 @@
...
     "eslint-plugin-react": "~6.9.0",
+    "flow-bin": "^0.52.0",
     "jshint": "~2.9.1",
     "jshint-loader": "~0.8.3",
-    "webpack": "^2.6.1"
+    "webpack": "^3.5.1"
   },
   "dependencies": {
-    "react-lite": "0.15.30"
+    "immutable": "^3.8.1",
```
Comment (kahowell) [inline cockpit/package.json]: Let's not pull in all these dependencies just yet (instead, let's pull them in as we're starting to use them). Also, they should probably be added as dev dependencies.

**Comment by kahowell** [comment]
Closing in favor of #1687 (some of this code incorporated there :smile:).

**Pull request closed without merging**